cmd/identity/identity.go

```diff
@@ -47,3 +79,59 @@ func cmdNewID(cmd *cobra.Command, args []string) (err error) {
 	}
 	return provider.ErrSetup.New("identity file(s) exist: %s", s)
 }
+
+func cmdLeafExtensions(cmd *cobra.Command, args []string) (err error) {
+	fi, err := leafExtCfg.Identity.Load()
+	if err != nil {
+		return err
+	}
+
+	return printExtensions(fi.Leaf.Raw, fi.Leaf.ExtraExtensions)
+}
+
+func cmdRevokeLeaf(cmd *cobra.Command, args []string) (err error) {
+	ca, err := revokeLeafCfg.CA.Load()
+	if err != nil {
+		return err
+	}
+	originalIdent, err := revokeLeafCfg.Identity.Load()
+	if err != nil {
+		return err
+	}
+
+	updatedIdent, err := ca.NewIdentity()
+	if err != nil {
+		return err
+	}
+
+	if err := peertls.AddRevocationExt(ca.Key, originalIdent.Leaf, updatedIdent.Leaf); err != nil {
+		return err
+	}
+
+	// NB: backup original cert
+	var backupCfg provider.IdentityConfig
+	certPathExt := filepath.Ext(revokeLeafCfg.Identity.CertPath)
+	certPath := revokeLeafCfg.Identity.CertPath
+	certBase := certPath[:len(certPath)-len(certPathExt)]
+	if err != nil {
+		return err
+	}
+	backupCfg.CertPath = fmt.Sprintf(
+		"%s.%s%s",
+		certBase,
+		strconv.Itoa(int(time.Now().Unix())),
+		certPathExt,
+	)
```

Comment:
This logic of generating the path of the backup certificate seems to be the same as in the `ca revoke` commands. Perhaps, it makes sense to extract it in a helper functions and share it in both commands.